### PR TITLE
chore: Opening response pane by default on query creation and for page load queries

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Opening response pane by default on query creation and for page load queries

Fixes [#37216](https://github.com/appsmithorg/appsmith/issues/37216)

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11723600626>
> Commit: 98db89cea1891e15fa5881503fa34a26d42cfd76
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11723600626&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Thu, 07 Nov 2024 13:48:18 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `PluginActionResponse` component for improved functionality and dynamic response handling.
	- Introduced a new state variable to manage response tab visibility on initial load.
	- Added logic to automatically open the response and schema tabs based on action responses.

- **Bug Fixes**
	- Improved control flow and error handling for displaying responses based on action success.

### CODE REVIEW

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx [hunk 1]
```diff
@@ -8,17 +8,76 @@ import { getPluginActionDebuggerState } from "../../store";
...
 
+  const [showResponseOnFirstLoad, setShowResponseOnFirstLoad] =
+    useState<boolean>(false);
+
+  const { responseDisplayFormat } =
+    actionResponseDisplayDataFormats(actionResponse);
+
+  // These useEffects are used to open the response tab by default for page load queries
+  // as for page load queries, query response is available and can be shown in response tab
+  useEffect(() => {
```
Comment (hetunandu) [inline app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:38]: Add a name to this use effect

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx [hunk 2]
```diff
@@ -8,17 +8,76 @@ import { getPluginActionDebuggerState } from "../../store";
...
+          selectedTab: DEBUGGER_TAB_KEYS.SCHEMA_TAB,
+        }),
+      );
+    }
+  }, [showSchema, selectedTab, dispatch]);
+
+  // When multiple page load queries exist, we want to response tab by default for all of them
+  // Hence this useEffect will reset showResponseOnFirstLoad flag used to track whether to show response tab or not
+  useEffect(() => {
+    if (action?.id) {
```
Comment (hetunandu) [inline app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:76]: No need to optional operator

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx [hunk 3]
```diff
@@ -8,17 +8,76 @@ import { getPluginActionDebuggerState } from "../../store";
...
+  const { responseDisplayFormat } =
+    actionResponseDisplayDataFormats(actionResponse);
+
+  // These useEffects are used to open the response tab by default for page load queries
+  // as for page load queries, query response is available and can be shown in response tab
+  useEffect(() => {
+    // actionResponse and responseDisplayFormat is present only when query has response available
+    if (
+      responseDisplayFormat &&
+      !!responseDisplayFormat?.title &&
```
Comment (hetunandu) [inline app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:42]: Are these needed? Would they be available when the action has a failed response?

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx [hunk 4]
```diff
@@ -8,17 +8,76 @@ import { getPluginActionDebuggerState } from "../../store";
...
+  const pluginRequireDatasource = doesPluginRequireDatasource(plugin);
+
+  const showSchema = useShowSchema(plugin?.id || "") && pluginRequireDatasource;
 
   // TODO combine API and Query Debugger state
   const { open, responseTabHeight, selectedTab } = useSelector(
     getPluginActionDebuggerState,
   );
 
+  const [showResponseOnFirstLoad, setShowResponseOnFirstLoad] =
```
Comment (hetunandu) [inline app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:30]: Is this a state that indicates a response tab is selected by default or it should be selected by default? The naming suggests the later but usage suggests the former

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx [hunk 5]
```diff
@@ -8,17 +8,76 @@ import { getPluginActionDebuggerState } from "../../store";
...
+          open: true,
+          selectedTab: DEBUGGER_TAB_KEYS.SCHEMA_TAB,
+        }),
+      );
+    }
+  }, [showSchema, selectedTab, dispatch]);
+
+  // When multiple page load queries exist, we want to response tab by default for all of them
+  // Hence this useEffect will reset showResponseOnFirstLoad flag used to track whether to show response tab or not
+  useEffect(() => {
```
Comment (hetunandu) [inline app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:75]: I do not understand how this use effect is affecting the above comment

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx:38]: add a name as in?

Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx]: done

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx (hunk 2 above)
Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx]: done

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx (hunk 3 above)
Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx]: This is required and basically works only when the response is a success. This is how it works in the normal flow as well.

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx (hunk 4 above)
Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx]: removed it

File: app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx (hunk 5 above)
Comment (ankitakinger) [reply app/client/src/PluginActionEditor/components/PluginActionResponse/PluginActionResponse.tsx]: removed it

**Review by hetunandu — APPROVED** [review]

**Pull request merged**